lib/crawler.js

```diff
@@ -1241,11 +1234,20 @@ Crawler.prototype.handleResponse = function(queueItem, response, timeCommenced)
     crawler.emit("fetchheaders", queueItem, response);
 
     // Ensure response length is reasonable...
-    responseLength =
-        responseLength > 0 ? responseLength : crawler.maxResourceSize;
-
+    responseLength = responseLength > 0 ? responseLength : crawler.maxResourceSize;
     queueItem.stateData.contentLength = responseLength;
 
+    function emitFetchComplete(responseBody, decompressedBuffer) {
+        responseBody = crawler.decodeResponses ? crawler.decodeBuffer(responseBody, stateData.contentType) : responseBody;
```

Comment:
Last time I ran it it didn't have great support for SHIFT-JIS and EUC-JP, but that might be better now.